src/libmongoc/src/mongoc/mcd-time.h

```diff
@@ -132,10 +205,19 @@ mcd_later (mcd_time_point from, mcd_duration delta)
 static inline mcd_duration
 mcd_time_difference (mcd_time_point then, mcd_time_point from)
 {
-   BSON_ASSERT (!_mcd_i64_sub_would_overflow (then._rep, from._rep));
-   int64_t diff = then._rep - from._rep;
-   // Our time_point encodes the time using a microsecond counter.
-   return mcd_microseconds (diff);
+   if (_mcd_i64_sub_would_overflow (then._rep, from._rep)) {
+      if ((then._rep < 0) == (from._rep < 0)) {
```

Comment:
Should this be `if (from._rep < 0) {`?

Positive overflow could only occur if subtracting a negative.

Reply:
You are correct. I was thinking multiplication.